src/sync/arc.rs

```diff
@@ -25,8 +25,30 @@ impl<T> Arc<T> {
     }
 
     /// Returns the inner value, if the `Arc` has exactly one strong reference.
-    pub fn try_unwrap(_this: Arc<T>) -> Result<T, Arc<T>> {
-        unimplemented!();
+    pub fn try_unwrap(this: Arc<T>) -> Result<T, Arc<T>> {
+        if this.obj.get_mut(location!()) {
```

Comment:
nit, take it or leave it: i have a personal preference for changing this from an `if/`else` to an `if` with an early return:
```suggestion
        if !this.obj.get_mut(location!()) {
             return Err(this);
        }
```
so that the rest of the code isn't indented inside of the `if` block...but, this is just a personal style preference.